### PR TITLE
Detection of availability of VirtualBox provider #11929

### PR DESCRIPTION
**Pull request by hpaskalev**
Fixes check for usability of VirtualBox provider upon initialization [Fix for #11929 issue](https://github.com/hashicorp/vagrant/issues/11929)

### CODE REVIEW

File: plugins/providers/virtualbox/provider.rb [hunk 1]
```diff
@@ -23,7 +23,8 @@ def self.usable?(raise_error=false)
       rescue Vagrant::Errors::VirtualBoxInvalidVersion,
              Vagrant::Errors::VirtualBoxNotDetected,
              Vagrant::Errors::VirtualBoxKernelModuleNotLoaded,
-             Vagrant::Errors::VirtualBoxInstallIncomplete
+             Vagrant::Errors::VirtualBoxInstallIncomplete,
+             Vagrant::Errors::VBoxManageNotFoundError
```
Comment (soapy1) [inline plugins/providers/virtualbox/provider.rb:27]: This looks great 🎉 
Could you also add a test for this. The tests for this `usable?` method exist in [test/unit/plugins/providers/virtualbox/provider_test.rb](https://github.com/hashicorp/vagrant/blob/master/test/unit/plugins/providers/virtualbox/provider_test.rb#L27). There are some pretty solid [examples]https://github.com/hashicorp/vagrant/blob/master/test/unit/plugins/providers/virtualbox/provider_test.rb#L35) of what the test should do.

**Review by soapy1 — CHANGES_REQUESTED** [review]
Thanks for the contribution! Looks good 😄

**Comment by hpaskalev** [comment]
![Uploading 95905367...]

File: plugins/providers/virtualbox/provider.rb (hunk 1 above)
Comment (hpaskalev) [reply plugins/providers/virtualbox/provider.rb:27]: Will do!

Comment (hpaskalev) [reply plugins/providers/virtualbox/provider.rb:27]: Done. Test scenario is added.

**Review by soapy1 — APPROVED** [review]
🎉

**Pull request merged**